CARDS-1164 (Smaller form headers on smaller screens) amendment: Moved the breadcrumb action to the RIGHT of the sticky header

marta- opened the pull request:
This positioning of the resource header actions should be compared to what is proposed in #728 and to what is already in `dev` via #723 , and one of the 3 must be chosen.

marta- committed:
CARDS-1164: Smaller form headers on smaller screens
Moved the breadcrumb action to the right of the sticky header

sdumitriu reviewed (approved):
Works well, and I like it more than on the left.

sashaandjic reviewed (approved):
I also prefer this one to the LEFT.
I would like to suggest using the same size of the breadcrumb action and trash can buttons.
Stretching the title and aligning `SUBJECT `/ , `Patient xyz` and `ENTERED BY... ` with DATES below, would improve the look as well.
![image](https://user-images.githubusercontent.com/23243244/128962921-69fbe760-95c3-4c59-b22f-ceb1a2b59a92.png)
![image](https://user-images.githubusercontent.com/23243244/128962983-80709f47-d524-4be7-8081-81e996e0accd.png)

marta- committed:
CARDS-1164: Smaller form headers on smaller screens
UI consistency improvements:
* The resource header action menu/button remains large as the header "shrinks" when scrolling up
* The action button in the subject page has a botder to resemble more the form menu

marta- committed:
CARDS-1164: Smaller form headers on smaller screens
Improved content alignment and font sizes on the subject page

marta- committed:
[misc] On the subject page, inform the user when there's no data to display instead of simply showing an empty box.

sashaandjic commented:
tested - perfect!

sdumitriu reviewed (commented):
In Firefox the title is not vertically centered:

![image](https://user-images.githubusercontent.com/88663/129255836-14401846-ed5d-449f-8876-20416daa850f.png)

sashaandjic commented:
confirmed: alignment is a bit off on both subject and form pages in Firefox
![image](https://user-images.githubusercontent.com/23243244/129269629-634bdcc8-e894-4110-a3af-6f8d72f1bd73.png)

marta- commented:
> In Firefox the title is not vertically centered:

Fixed in 3fed284.

marta- committed:
CARDS-1164: Smaller form headers on smaller screens
Addressing review comments: fixed the action fit inside the sticky header in Firefox

sashaandjic reviewed (approved):
retested - all good
